portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,4 +90,40 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     // the response. Should consider whether or not this issue should be handled
     // on the server side or not.
   }
-}
+  private void refreshData(HttpServletRequest request,
+                           HttpServletResponse response) {
+
+    int queryLimit;
+
+    String requestLimit = request.getParameter("limit");
+
+    try {
+      // Convert the input to an integer.
+      queryLimit = Integer.parseInt(requestLimit);
+      System.out.println("parse successful");
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + requestLimit +
+                         " use default");
+      queryLimit = 20;
+    }
+
+    // Reset the data.
+    userData.clear();
+    System.out.println(queryLimit);
+    FetchOptions options = FetchOptions.Builder.withLimit(queryLimit);
+    Query recentComments =
+        new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    List<Entity> results = storeData.prepare(recentComments).asList(options);
+    for (Entity entity : results) {
+      System.out.println("This is running");
+      String user = entity.getProperty("user").toString();
+      String comment = entity.getProperty("comment").toString();
+      // TODO(morleyd): Change implementation to use a hash to lower time
+      // complexity, need to figure out how this interacts with Gson.
+      User userComment = new User(user, comment);
+      if (!userData.contains(userComment)) {
+        userData.add(userComment);
+      }
```

Comment:
Note that this has the effect of de-duplicating multiple comments with the same username and content. I'm not sure if that's your intent or not; it might be nice to have a comment explaining if so.